### PR TITLE
add more input_output_types found from breaking scripts

### PR DESCRIPTION
**Pull request by fdncred**
# Description

This PR fixes some problems I found in scripts by adding some additional input_output_types.

Here's a list of nushell scripts that it fixed. Look for `# broke here:` below.

This PR fixes 3, 4, 6, 7 by adding additional input_output_types. 1 was fixed by changing the script. 2. just doesn't work anymore because mkdir return type has changed. 5, is a problem with the script, the datatype for `...rest` needed to be removed.

```nushell
# 1.
def terminal-size [] {
    let sz = (input (ansi size) --bytes-until 'R')
    # $sz should look like this
    # Length: 9 (0x9) bytes | printable whitespace ascii_other non_ascii
    # 00000000:   1b 5b 33 38  3b 31 35 30  52                         •[38;150R
    let sz_len = ($sz | bytes length)

    # let's skip the esc[ and R
    let r = ($sz | bytes at 2..($sz_len - 2) | into string)

    # $r should look like 38;150
    # broke here: because $r needed to be a string for split row
    let size = ($r | split row ';')

    # output in record syntax
    {
        rows: ($size | get 0)
        columns: ($size | get 1)
    }
}

# 2.
# make and cd to a folder
def-env mkcd [name: path] {
    # broke here: but apparently doesn't work anymore
    # It looks like  mkdir returns nothing where it used to return a value
    cd (mkdir $name -v | first) 
}

# 3.
# changed 'into datetime'
def get-monday [] {
  (seq date -r --days 7 |
  # broke here: because into datetime didn't support list input
   into datetime | 
   where { |e| 
   ($e | date format %u) == "1" }).0 | 
   date format "%Y-%m-%d"
}

# 4.
# Delete all branches that are not in the excepts list
# Usage: del-branches [main]
def del-branches [
    excepts:list  # don't delete branch in the list
    --dry-run(-d) # do a dry-run
 ] {
    let branches = (git branch | lines | str trim)
    # broke here: because str replace didn't support list<string>
    let remote_branches = (git branch -r | lines | str replace '^.+?/' '' | uniq)
    if $dry_run {
        print "Starting Dry-Run"
    } else {
        print "Deleting for real"
    }
    $branches | each {|it|
        if ($it not-in $excepts) and ($it not-in $remote_branches) and (not ($it | str starts-with "*")) {
            # git branch -D $it
            if $dry_run {
                print $"git branch -D ($it)"
            } else {
                print $"Deleting ($it) for real"
                #git branch -D $it
            }
        }
    }
}

# 5.
# zoxide script
def-env __zoxide_z [...rest] {
  # `z -` does not work yet, see https://github.com/nushell/nushell/issues/4769
  # broke here: 'append doesn't support string input'
  let arg0 = ($rest | append '~').0
  # broke here: 'length doesn't support string input' so change `...rest:string` to `...rest`
  let path = if (($rest | length) <= 1) and ($arg0 == '-' or ($arg0 | path expand | path type) == dir) {
    $arg0
  } else {
    (zoxide query --exclude $env.PWD -- $rest | str trim -r -c "\n")
  }
  cd $path
}

# 6.
def a [] { 
    let x = (commandline)
    if ($x | is-empty) { return }
    # broke here: because commandline was previously only returning Type::Nothing
    if not ($x | str starts-with "aaa") { print "bbb" }
}

# 7.
# repeat a string x amount of times
def repeat [arg: string, dupe: int] {
  # broke here: 'command does not support range input'
  0..<$dupe | reduce -f '' {|i acc| $acc + $arg}
}
```

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

File: crates/nu-command/src/conversions/into/datetime.rs [hunk 1]
```diff
@@ -67,6 +67,7 @@ impl Command for SubCommand {
         .input_output_types(vec![
             (Type::Int, Type::Date),
             (Type::String, Type::Date),
+            (Type::List(Box::new(Type::String)), Type::List(Box::new(Type::Date))),
```
Comment (sholderbach) [inline crates/nu-command/src/conversions/into/datetime.rs:70]: Would [`.vectorizes_over_list()`](https://docs.rs/nu-protocol/latest/nu_protocol/struct.Signature.html#method.vectorizes_over_list) suffice here? Or does it not work for the `Int -> Date` signature like that.

File: crates/nu-command/src/strings/str_/replace.rs [hunk 2]
```diff
@@ -37,6 +37,10 @@ impl Command for SubCommand {
             .input_output_types(vec![
                 (Type::String, Type::String),
                 (Type::Table(vec![]), Type::Table(vec![])),
+                (
+                    Type::List(Box::new(Type::String)),
+                    Type::List(Box::new(Type::String)),
+                ),
```
Comment (sholderbach) [inline crates/nu-command/src/strings/str_/replace.rs:43]: Mhh maybe not and there is a bug in that.

File: crates/nu-command/src/conversions/into/datetime.rs (hunk 1 above)
Comment (fdncred) [reply crates/nu-command/src/conversions/into/datetime.rs:70]: I remove that new i/o line and added `.vectorizes_over_list()` and still get the same error.
```nushell
seq date -r --days 7 | into datetime
Error: nu::parser::input_type_mismatch

  × Command does not support list<string> input.
   ╭─[entry #1:1:1]
 1 │ seq date -r --days 7 | into datetime
   ·                        ──────┬──────
   ·                              ╰── command doesn't support list<string> input
```

File: crates/nu-command/src/strings/str_/replace.rs (hunk 2 above)
Comment (fdncred) [reply crates/nu-command/src/strings/str_/replace.rs:43]: I'm not following your meaning. Are you saying that `str replace` shouldn't take a list and return a list?

Comment (sholderbach) [reply crates/nu-command/src/strings/str_/replace.rs:43]: I saw that you had to add the list types despite `.vectorizes_over_list()` so it seems like it doesn't so what it says.

Comment (fdncred) [reply crates/nu-command/src/strings/str_/replace.rs:43]: ok, ya. i'm not sure what the purpose to vectorizes_over_lists is then. i tried it in your previous comment and it didn't seem to do anything. maybe i'm just misunderstanding what it's purpose is.

**Pull request merged**